src/Common/ZooKeeper/ZooKeeper.cpp

```diff
@@ -783,7 +785,21 @@ std::future<Coordination::GetResponse> ZooKeeper::asyncGet(const std::string & p
             promise->set_value(response);
     };
 
-    impl->get(path, std::move(callback), {});
+    impl->get(path, std::move(callback), watch_callback);
+    return future;
+}
+
+std::future<Coordination::GetResponse> ZooKeeper::tryAsyncGetNoThrow(const std::string & path, Coordination::WatchCallback watch_callback)
+{
+    auto promise = std::make_shared<std::promise<Coordination::GetResponse>>();
+    auto future = promise->get_future();
+
+    auto callback = [promise, path](const Coordination::GetResponse & response) mutable
```

Comment:
Is `path` is really needed here? Seems like this is another dangling reference.

Reply:
all non `&name` variables captured by copy, even if they are references. But yes, this one is redundant.